### PR TITLE
Add `RELEASE_NOTES.md`

### PR DESCRIPTION
**Pull request by foundrytom**
Required for #383.

### CODE REVIEW

File: contributing/CHANGES.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
-# Recording user facing changes in CHANGES.md
```
Comment (feltech) [inline contributing/CHANGES.md:1]: [question] Is this file's name remaining the same?

Comment (feltech) [inline contributing/CHANGES.md:1]: [fix] The example under "Formatting" below still has "Changes" in the title (maybe the title is irrelevant to that section and can be removed)

**Review by feltech — CHANGES_REQUESTED** [review]

File: contributing/CHANGES.md (hunk 1 above)
Comment (foundrytom) [reply contributing/CHANGES.md:1]: If that seems ok? Was thinking it referred to the 'first' use of changes, rather than the second.

Comment (foundrytom) [reply contributing/CHANGES.md:1]: Good spot, thanks! Fixed [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/386/commits/0a9f7bf51e7070e80142554ac20bd7f81d9c4515).

**Review by feltech — APPROVED** [review]
LGTM, squash and merge when ready

**Pull request merged**